### PR TITLE
Change Data Type for Several Variables

### PR DESCRIPTION
**Pull request by kaikawaliu**
## Description

Changed variables:

- Transaction:
version: uint256_t -> uint32_t
nonce: uint256_t -> uint64_t
gasLimit: uint256_t -> uint64_t
amount: uint256_t -> uint128_t
gasPrice: uint256_t -> uint128_t

- Account:
nonce: uint256_t -> uint64_t
balance: uint256_t -> uint128_t

- TxBlockHeader:
gasLimit: uint256_t -> uint64_t
gasUsed: uint256_t -> uint64_t
timestamp: uint256_t -> uint64_t
rewards: uint256_t -> uin128_t

- MicroBlockHeader:
gasLimit: uint256_t -> uint64_t
gasUsed: uint256_t -> uint64_t
timestamp: uint256_t -> uint64_t
rewards: uint256_t -> uin128_t

- DSBlockHeader
timestamp: uint256_t -> uint64_t
gasPrice: uint256_t -> uint128_t

- VCBlockHeader
timestamp: uint256_t -> uint64_t

- FallbackBlockHeader
timestamp: uint256_t -> uint64_t

- [x] Also changed the function Account::GetAddressForContract

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test with rejoin (including retrieving data from db)

- [x] 200 nodes small-scale cloud test (commit: 7bfc140, 58d2d9b)

### CODE REVIEW

**Comment by KaustubhShamshery** [comment]
Btw can you reference the issue here.

**Comment by kaikawaliu** [comment]
> Btw can you reference the issue here.

Can't remember whether there is a issue for this

**Comment by KaustubhShamshery** [comment]
#774

File: tests/Crypto/Test_Schnorr.cpp [hunk 1]
```diff
@@ -135,7 +135,7 @@ BOOST_AUTO_TEST_CASE(test_sign_verif) {
   pair<PrivKey, PubKey> keypair = schnorr.GenKeyPair();
 
   /// 1 MB message
-  const unsigned int message_size = 1048576;
+  const unsigned int message_size = 201;
```
Comment (ShengguangXiao) [inline tests/Crypto/Test_Schnorr.cpp]: The comment and data size is not match

Comment (kaikawaliu) [reply tests/Crypto/Test_Schnorr.cpp]: reverted the change

**Comment by bb111189** [comment]
Will need to resolve conflicts

**Pull request merged**